src/extensions/sessionrecording-utils.ts

```diff
@@ -82,3 +133,103 @@ export function truncateLargeConsoleLogs(_event: eventWithTime) {
     }
     return _event
 }
+
+export class MutationRateLimiter {
+    private bucketSize = 100
+    private refillRate = 10
+    private mutationBuckets: Record<string, number> = {}
+    private loggedTracker: Record<string, boolean> = {}
+
+    constructor(
+        private readonly rrweb: rrwebRecord,
+        private readonly options: {
+            bucketSize?: number
+            refillRate?: number
+            onBlockedNode?: (id: number, node: Node | null) => void
+        } = {}
+    ) {
+        this.refillRate = this.options.refillRate ?? this.refillRate
+        this.bucketSize = this.options.bucketSize ?? this.bucketSize
+        setInterval(() => {
+            this.refillBuckets()
+        }, 1000)
+    }
+
+    private refillBuckets = () => {
+        Object.keys(this.mutationBuckets).forEach((key) => {
+            this.mutationBuckets[key] = this.mutationBuckets[key] + this.refillRate
+
+            if (this.mutationBuckets[key] >= this.bucketSize) {
+                delete this.mutationBuckets[key]
```

Comment:
If we want to go into  bike-shedding mode

`delete` is way, way slower than setting a key to undefined or null

https://jsbench.me/b7lki4u1j5/1

Might not matter by then we call this relatively frequently

It should be ok since the size of the object is pretty constrained but 🤷

Reply:
Can do

Reply:
Although... If we're not deleting then we have an ever growing object of keys...

Reply:
Yeah, if we weren't calling this on a timer then I think it'd be worth it but since `delete` can do 8.5M operations a second.... If we have 8.5M nodes in the object I guess our problem isn't how fast we can delete them 🙈 

Told you it was bike-shedding